zerver/views/upload.py

```diff
@@ -88,12 +93,16 @@ def serve_local(
     attachment = download or mimetype not in INLINE_MIME_TYPES
 
     if settings.DEVELOPMENT:
+        # In development, we do not have the nginx server to offload
+        # the response to; serve it directly ourselves.
         response: HttpResponseBase = FileResponse(open(local_path, "rb"), as_attachment=attachment)
-    else:
-        response = HttpResponse()
-        response["X-Accel-Redirect"] = quote(f"/internal/uploads/{path_id}")
-        del response["Content-Type"]
-        patch_disposition_header(response, local_path, attachment)
+        patch_cache_control(response, private=True, immutable=True)
+        return response
+
+    response = HttpResponse()
+    response["X-Accel-Redirect"] = quote(f"/internal/uploads/{path_id}")
+    del response["Content-Type"]
```

Comment:
Should this `del` line have a comment explaining why we do it? Or possibly a slightly longer one explaining at a high level we're doing with headers more generally in this block would be better.

Reply:
I've done some re-jiggering to factor this out into a function, sooner in the series, and document why.